core/dataset.cpp

```diff
@@ -1151,10 +1151,14 @@ Variable histogram(const DataConstProxy &sparse,
   }
   std::copy(result.values<double>().begin(), result.values<double>().end(),
             result.variances<double>().begin());
-  return result;
+
+  std::map<Dim, Variable> coords;
+  coords.emplace(dim, binEdges);
+  return {std::move(result), std::move(coords),
+          std::map<std::string, Variable>()};
```

Comment:
Should we also include all other coords and labels from the input(except for the sparse ones)? I am not sure what would be the most consistent / expected way? I think we should?